File: Fortran/ex2/predict.f90
	subroutine predict(M, N, X, theta, gg)

	implicit none 
	
	integer::M, N
	double precision,dimension(M,N)::X	
	double precision,dimension(M)::gg
	double precision,dimension(N)::theta
	double precision::TX, sigmoid
	integer::i, k, j
	
	
	do i=1, M	
	
	          TX = 0
		    do j=1, N
			TX = TX + X(i,j)*theta(j)
		    enddo
		    
		    gg(i) = sigmoid(TX)	
		    	 				 
	enddo

	
		
	 end subroutine predict

File: C/ex2/predict.c

double sigmoid(double z);

int predict(int M, int N, double theta[N], double X[M][N], double g[M])
{
	double TX ;
	int i, j;
	
	for(i=0; i < M; i++){
           TX = 0;
	   for(j=0; j < N; j++)
	       TX = TX + theta[j]*X[i][j];
	   g[i] = sigmoid(TX);
	}
	   			
	return 0 ;		
}
